Explorer/Assets/DCL/SDKComponents/Tween/Systems/TweenUpdaterSystem.cs

```diff
@@ -0,0 +1,220 @@
+using DG.Tweening;
+using ECS.Abstract;
+using ECS.LifeCycle;
+using ECS.Unity.Groups;
+using ECS.Unity.Transforms.Components;
+using ECS.Unity.Tween.Components;
+using System.Collections.Generic;
+using UnityEngine;
+using static DCL.ECSComponents.EasingFunction;
+using static DG.Tweening.Ease;
+
+namespace ECS.Unity.Tween.Systems
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    [UpdateAfter(typeof(TweenLoaderSystem))]
+    [LogCategory(ReportCategory.TWEEN)]
+    [ThrottlingEnabled]
+    public partial class TweenUpdaterSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private readonly WorldProxy globalWorld;
+        private Tweener currentTweener;
+
+        public TweenUpdaterSystem(World world, WorldProxy globalWorld) : base(world)
+        {
+            this.globalWorld = globalWorld;
+        }
+
+        protected override void Update(float t)
+        {
+            SetupTweenSequenceQuery(World);
+        }
+
+        public void FinalizeComponents(in Query query) { }
+
+        private TweenStateStatus GetTweenState(float currentTime, bool isPlaying)
+        {
+            if (!isPlaying) { return TweenStateStatus.TsPaused; }
+
+            return currentTime.Equals(1f) ? TweenStateStatus.TsCompleted : TweenStateStatus.TsActive;
+        }
+
+        [Query]
+        [All(typeof(SDKTweenComponent))]
+        private void SetupTweenSequence(ref SDKTweenComponent sdkTweenComponent, ref TransformComponent transformComponent)
+        {
+            if (!sdkTweenComponent.IsDirty)
+            {
+                //We also need to update transform positions
+                float currentTime = sdkTweenComponent.tweener.ElapsedPercentage();
+                TweenStateStatus newState = GetTweenState(currentTime, sdkTweenComponent.isPlaying);
+
+                if (newState != sdkTweenComponent.TweenStateStatus)
+                {
+                    sdkTweenComponent.TweenStateStatus = newState;
+                    sdkTweenComponent.IsTweenStateDirty = true;
+                }
+
+                if (sdkTweenComponent.isPlaying && !sdkTweenComponent.CurrentTime.Equals(currentTime))
+                {
+                    sdkTweenComponent.CurrentTime = currentTime;
+                    sdkTweenComponent.IsTweenStateDirty = true;
+                }
+            }
+            else
+            {
+                PBTween tweenModel = sdkTweenComponent.CurrentTweenModel;
+                bool isPlaying = !tweenModel.HasPlaying || tweenModel.Playing;
+                sdkTweenComponent.isPlaying = isPlaying;
+
+                Transform entityTransform = transformComponent.Transform;
+                float durationInSeconds = tweenModel.Duration / 1000;
+
+                currentTweener = sdkTweenComponent.tweener;
+
+                // There may be a tween running for the entity transform, even though internalComponentModel.tweener
+                // is null, e.g: during preview mode hot-reload.
```

Comment:
Replace with `// There may be a tween running for the entity transform, e.g: during preview mode hot-reload.` since `internalComponentModel` is a concept of the old explorer and doesn't exist in the explorer alpha

Reply:
done!